pkg/asset/agent/agentconfig/agenthosts.go

```diff
@@ -201,8 +201,17 @@ func (a *AgentHosts) validateRendezvousIPNotWorker(rendezvousIP string, hosts []
 
 	if rendezvousIP != "" {
 		for i, host := range hosts {
+			if host.Role != workerRole {
+				continue
+			}
 			hostPath := field.NewPath("Hosts").Index(i)
-			if strings.Contains(string(host.NetworkConfig.Raw), rendezvousIP) && host.Role == workerRole {
+			hostIPs, err := agentAsset.GetAllHostIPs(host.NetworkConfig)
+			if err != nil {
+				allErrs = append(allErrs, field.Invalid(hostPath, host.NetworkConfig, err.Error()))
+				continue
+			}
+			_, found := hostIPs[rendezvousIP]
```

Comment:
nice!

Reply:
Btw, I was expecting to find at least a unit test to verify that the expected error didn't happen again?

Reply:
Added unit test. Confirmed that without this fix the test will fail with             	            	```
Hosts[1].Host: Forbidden: Host test-2 has role 'worker' and has the rendezvousIP assigned to it. The rendezvousIP must be assigned to a control plane host.
            	            	invalid Hosts configuration
```